### PR TITLE
Editor

### PR DESCRIPTION
**Pull request by shundroid**
すみません、初回で、ちょっと量が多くなってしまいます。
Editor に書いて、保存できるところまでをこのPRでやろうと思います。

This change is [Reviewable](https://reviewable.io/reviews/comozilla/onigo-controller/4)

### CODE REVIEW

**Comment by dadaa** [comment]
Review status: 0 of 17 files reviewed at latest revision, 6 unresolved discussions.

---

_[index.html, line 14 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk2g7Jdisdmg9jspRI:-KLk2g7Jdisdmg9jspRJ:-803652092) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/index.html#L14)):_

> ``` HTML
>     </div>
>   </div>
>   <div id="main">
> ```

意味的に main でしたら，
<main>
を使ってみてはいかがでしょうか．
https://developer.mozilla.org/ja/docs/Web/HTML/Element/main

---

_[webpack.config.js, line 1 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk3MZVxPvsiIac68vd:-KLk3MZWqxv9ZuS8pWgY:-1784724348) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/webpack.config.js#L1)):_

> ``` JavaScript
> const webpack = require("webpack");
> ```

この js の全ての行末に半角スペースが入っているようです．
他のファイルも同様なところがあるようなので，そこも消しましょう．

---

_[js/block-manager.js, line 12 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk4FeAUiupvbWmU5wW:-KLk4FeBlYnvHGFWDbQT:-973749178) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block-manager.js#L12)):_

> ``` JavaScript
>   this.editor = editor;
> 
>   this.blocks = [];
> ```

this.blocks = document.querySelectorAll("[data-block-index]");
で，いかがでしょう．そのあと，これをループして処理をする．
これで，const blockCount = 9; の行は削除できそうです．

---

_[js/block-manager.js, line 13 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk3dvJruw5bHCPxFzg:-KLk3dvJruw5bHCPxFzh:131417675) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block-manager.js#L13)):_

> ``` JavaScript
> 
>   this.blocks = [];
>   for (var i = 0; i < blockCount; i++) {
> ```

const など ES6 を利用するのであれば，var の代わりに let を使ってはいかがでしょうか．他のところも同じように．

---

_[js/block.js, line 9 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk5npNFR3_XIVoAT5D:-KLk5npOUepfVoj_LFSf:1346858714) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block.js#L9)):_

> ``` JavaScript
>   this.enable = true;
>   this.blockManager = blockManager;
>   this.mode = "making";
> ```

モードの変数は enum 的なものを使いたいと思うので，これも ES6 の Symbol などを試してみてはいかがでしょう
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Symbol

---

_[js/sphero-client.js, line 51 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk7_FbWIxoAVtJyQ5k:-KLk7_FcOT48yp1BtZSk:-667742850) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/sphero-client.js#L51)):_

> ``` JavaScript
>     });
>     this.orb.listenCustomMessage("availableCommandsCount", (data) => {
>       console.log(data);
> ```

これはデバッグ用？

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 25 files reviewed at latest revision, 6 unresolved discussions.

---

_[js/block-manager.js, line 12 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk4FeAUiupvbWmU5wW:-KLkT0g3j7a9X0j_YtMx:-1635702748) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block-manager.js#L12)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> this.blocks = document.querySelectorAll("[data-block-index]");
> で，いかがでしょう．そのあと，これをループして処理をする．
> これで，const blockCount = 9; の行は削除できそうです．
> </details>
> そうします

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 26 files reviewed at latest revision, 6 unresolved discussions.

---

_[js/sphero-client.js, line 51 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk7_FbWIxoAVtJyQ5k:-KLkbc_AhMJC1VMSdSoH:833327008) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/sphero-client.js#L51)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> これはデバッグ用？
> </details>
> そうでした！

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 26 files reviewed at latest revision, 6 unresolved discussions.

---

_[js/block.js, line 9 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk5npNFR3_XIVoAT5D:-KLkbhdn2xTXaUjHpmSH:48136740) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block.js#L9)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> モードの変数は enum 的なものを使いたいと思うので，これも ES6 の Symbol などを試してみてはいかがでしょう
> https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Symbol
> </details>
> そうしました！

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 26 files reviewed at latest revision, 6 unresolved discussions.

---

_[js/block-manager.js, line 13 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk3dvJruw5bHCPxFzg:-KLkbtNxYHPurvbbEijd:1997776133) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/js/block-manager.js#L13)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> const など ES6 を利用するのであれば，var の代わりに let を使ってはいかがでしょうか．他のところも同じように．
> </details>
> ESlintを使って直しました。

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 26 files reviewed at latest revision, 6 unresolved discussions.

---

_[webpack.config.js, line 1 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk3MZVxPvsiIac68vd:-KLkbw1RuhmTJzmlWEqd:-27304601) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/webpack.config.js#L1)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> この js の全ての行末に半角スペースが入っているようです．
> 他のファイルも同様なところがあるようなので，そこも消しましょう．
> </details>
> あ、これ改行コードの問題だったっぽいです。

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

**Comment by shundroid** [comment]
Review status: 0 of 26 files reviewed at latest revision, 6 unresolved discussions.

---

_[index.html, line 14 [r1]](https://reviewable.io:443/reviews/comozilla/onigo-controller/4#-KLk2g7Jdisdmg9jspRI:-KLkc01NWHu5TO9N9o7j:-499146009) ([raw file](https://github.com/comozilla/onigo-controller/blob/324cc39fbd81f4ec499b6f4475fa3f510ffe3831/index.html#L14)):_
<details><summary>

<i>Previously, dadaa wrote…</i></summary>



> 意味的に main でしたら，
> <main>
> を使ってみてはいかがでしょうか．
> https://developer.mozilla.org/ja/docs/Web/HTML/Element/main
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/onigo-controller/4)_

File: index.html [hunk 1]
```diff
@@ -11,19 +11,49 @@
...
+    <button id="switch-mode-button">モードきりかえ</button>
+    <div id="making-mode-banner">
+      <span id="mode">つくるモード</span>
+      <button>ほぞん</button>
     </div>
-    <button id="calibration-button"></button>
-  </div>
+    <div id="playing-mode-banner">
+      <span id="mode">プレイモード</span>
+      <span id="hp">HP:100</span>
```
Comment (dadaa) [inline index.html]: reviewable がなんだか調子悪く，全てを表示できなかったので，直接ここでコメントしちゃいますね．

"HP:" の部分が固定なのでしたら，たとえば before 擬似要素を使うのも良いかと思います．
https://developer.mozilla.org/ja/docs/Web/CSS/::before

e.g.
HTML:
`<span id="hp">100</span>`
CSS:
`#hp:before { 
  content: "HP:";
}`

File: js/hp-board.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+  this.hp = 0;
+  eventPublisher.subscribe("hp", (hp) => {
+    this.hp = hp;
+    this.updateHP();
+  });
+  eventPublisher.publish("hp", 100);
+}
+
+HPBoard.prototype.updateHP = function() {
+  this.element.textContent = "HP:" + this.hp.toString();
```
Comment (dadaa) [inline js/hp-board.js]: before 擬似要素を使うように修正すると，ここの "HP:" もいらなくなりそうです．

File: js/main.js [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
+import MotionManager from "./motion-manager";
+import BlockManager from "./block-manager";
+import Editor from "./editor";
+
+// webpack
+import "../css/style.css";
+import "font-awesome";
+import "web-animations-js";
+
+let spheroStates;
```
Comment (dadaa) [inline js/main.js]: このあたりの変数は，ここに宣言する理由がありますかしら？
document.addEventListener("DOMContentLoaded", function() {
の内部だけだとダメ？

File: index.html (hunk 1 above)
Comment (shundroid) [inline index.html]: そうします！

File: js/main.js (hunk 3 above)
Comment (shundroid) [inline js/main.js]: 内部に入れます。

File: js/main.js [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
+import BlockManager from "./block-manager";
+import Editor from "./editor";
+
+// webpack
+import "../css/style.css";
+import "font-awesome";
+import "web-animations-js";
+
+document.addEventListener("DOMContentLoaded", function() {
+  let inactiveStatus = new InactiveStatus();
```
Comment (dadaa) [inline js/main.js]: いくつかの変数は宣言だけして使わないから，変数として持たなくてもいいかも
あと，使う変数( e.g. motionManager )などはここでは固定になりそうなので const で良いかな

Comment (shundroid) [inline js/main.js]: たしかにそうですね・・。そうします！

Comment (shundroid) [inline js/main.js]: 変更しました

File: css/style.css [hunk 5]
```diff
@@ -3,81 +3,149 @@ html {
...
+  display: inline-block;
+  font-size: 16px;
+  padding: 5px;
+  margin: 5px;
   border: none;
+  color: white;
+  border-radius: 10px;
+}
+
+footer #making-mode-banner span {
```
Comment (dadaa) [inline css/style.css]: #making-mode-banner span {
で，指定できていそうです．
この下に続くところも同様かな

Comment (shundroid) [inline css/style.css]: 変更しました！

**Comment by dadaa** [comment]
おつかれさまでした〜

**Comment by shundroid** [comment]
ありがとうございます！

**Pull request merged**